api/v1alpha1/vineyardruntime_types.go

```diff
@@ -0,0 +1,216 @@
+	// Default is `IfNotPresent`.
+	// +optional
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+
+	// NodeSelector is a selector to choose which nodes to launch the Vineyard component.
+	// E,g. {"disktype": "ssd"}
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// Ports used by Vineyard component.
+	// For Master, the default client port is 2379 and peer port is 2380.
+	// For Worker, the default rpc port is 9600.
+	// +optional
+	Ports map[string]int `json:"ports,omitempty"`
+
+	// Environment variables that will be used by vineyard component.
+	// For Master, refer to <a href="https://etcd.io/docs/v3.5/op-guide/configuration/">Etcd Configuration</a> for more info
+	// Default is not set.
+	// +optional
+	Env map[string]string `json:"env,omitempty"`
+
+	// VolumeMounts specifies the volumes listed in ".spec.volumes" to mount into the vineyard runtime component's filesystem.
+	// It is useful for specifying a persistent storage.
+	// Default is not set.
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+// MasterSpec defines the configurations for Vineyard Master component
+// which is also regarded as the Etcd component in Vineyard.
+// For more info about Vineyard, refer to <a href="https://v6d.io/">Vineyard</a>
+type MasterSpec struct {
+	// the component configurations for Vineyard Master
+	// +optional
+	VineyardCompTemplateSpec `json:",inline"`
+
+	// whether to deploy the Vineyard Master component
+	// Default is true
+	// If set to false, the Etcd component will not be deployed.
+	// Which means the Vineyard Worker component will use an external Etcd cluster.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// Resources contains the resource requirements and limits for the Vineyard Master.
+	// Default is not set.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+// WorkerSpec defines the configurations for Vineyard Worker component
+// which is also regarded as the Vineyardd component in Vineyard.
+// For more info about Vineyard, refer to <a href="https://v6d.io/">Vineyard</a>
+type WorkerSpec struct {
+	// the common configurations for Vineyard Worker
+	// +optional
+	VineyardCompTemplateSpec `json:",inline"`
+
+	// Configurable properties for Vineyard Worker
+	// Support the following properties.
+	//
+	//   vineyardd.reserve.memory: (Bool) where to reserve memory for vineyardd
+	//                             If set to true, the memory quota will be counted to the vineyardd rather than the application.
+	//   vineyardd.external.etcd.endpoint: (String) the endpoint of external etcd cluster
+	//
+	// E,g. {"vineyardd.external.etcd.endpoint": "etcd-svc.etcd-namespace.svc.cluster.local:2379/10.244.3.4:2379"}
+	//
+	// Default value is as follows.
+	//
+	//   vineyardd.reserve.memory: "true"
+	//
+	// +optional
+	Properties map[string]string `json:"properties,omitempty"`
```

Comment:
How about renaming properties to options? And I think master spec also need options which can provides flexibility for future.

Reply:
Make sense to me.